### DISCUSSION TITLE
[Copilot] Extension is uninstalled but still getting prompted to configure it.

### DISCUSSION THREAD

**Discussion by andreujuanc** [community/community#33479] [Copilot Conversations]
System:
 Windows 11, WSL2 + DevContainers

In some repos the extension is still "installed" within the DevContainer and I get the prompt
![image](https://user-images.githubusercontent.com/360052/190984793-2aed892e-0943-4f0b-b33b-7de275f50496.png)
![image](https://user-images.githubusercontent.com/360052/190984843-36c486c0-5750-4d02-a895-e677b7e059a7.png)

I tripple checked and the extension is not installed in `local` nor in `WSL`.  It's persisted even after I rebuild the container without cache.
And it is not in the `extensions[]` section of the `devcontainer.json`.

I'm opening this here because it hasn't happened with any other extension and I've been using WSL2+DevContainers for years.

**Comment by kimadeline** [comment]
Hi @andreujuanc, thank you for reaching out.

Do you have repro steps and/or a repo or a devcontainer.json to share so we can try to repro the issue?

**Reply by andreujuanc** [reply]
Sure,

One of the repos it happens:
```json
{
	"name": "El Dorado App",
	"context": "..",
	"dockerFile": "Dockerfile",
	"settings": {},
	"extensions": [
		"dart-code.flutter"
	],
	"runArgs": [
		"--cap-add=ALL",
		"--privileged",
		"-e DISPLAY",
		"-e WAYLAND_DISPLAY",
		"-e XDG_RUNTIME_DIR",
		"-e PULSE_SERVER"
	],
	"containerEnv": {
		"DISPLAY": "${localEnv:DISPLAY}",
		"WAYLAND_DISPLAY": "${localEnv:WAYLAND_DISPLAY}",
		"XDG_RUNTIME_DIR": "${localEnv:XDG_RUNTIME_DIR}",
		"PULSE_SERVER": "${localEnv:PULSE_SERVER}"
	},
	"mounts": [
		"source=/tmp/.X11-unix,target=/tmp/.X11-unix,type=bind",
		"source=/mnt/wslg,target=/mnt/wslg,type=bind"
	],
	"postStartCommand": "echo developer | sudo -S chown developer /dev/kvm",
	"remoteUser": "developer"
}
```

I rebuilt the container and it still happens.
![image](https://user-images.githubusercontent.com/360052/191573479-cc67096f-7b2b-4d32-9885-51a46e968c40.png)

I suspect that it was "injected" back when the extension was valid, and it's now baked in the container. So I went and rebuilt the container WITHOUT cache, and it's still showing:
![image](https://user-images.githubusercontent.com/360052/191574698-091278a0-26bb-44c0-8ac4-28c1388b6928.png)
![image](https://user-images.githubusercontent.com/360052/191574768-2f9047e3-3a96-446e-98d8-324fa71614d3.png)

**Reply by tilmann1** [reply]
The GitHub Copilot extension always needs to be installed manually (VS Code will never install it by default). This seems to indicate that this is a general issue related to how the container is built. Recommendation is to change the container configuration in a way that the Copilot extension is not installed in the first place. Thanks!

**Reply by tilmann1** [reply]
Side note: In your screenshot above the GitHub Copilot extension is definitely installed as indicated by the "228 ms" initialization time on the far right.

**Reply by andreujuanc** [reply]
it's installed but it's not anywhere to be found outside of the container. And yes, this might be an issue of the devContainers "cache" or something.

**Reply by kimadeline** [reply]
I'd suggest filing an issue on https://github.com/Microsoft/vscode-remote-release since this seems like an issue with your container, and we (as in, the Copilot extension) don't manage its installation process.

**Reply by andreujuanc** [reply]
Makes sense. Will do.

**Reply by andreujuanc** [reply]
@kimadeline FYI https://github.com/microsoft/vscode-remote-release/issues/7239

**Reply by kimadeline** [reply]
[Seems](https://github.com/microsoft/vscode-remote-release/issues/7239#issuecomment-1257247259) like you got it figured out after all 🙂
